modules/IOUtils.tla

```diff
@@ -10,6 +10,19 @@ IOSerialize(val, absoluteFilename, compress) == TRUE
 
 IODeserialize(absoluteFilename, compressed) == CHOOSE val : TRUE
 
+Serialize(value, dest, options) ==
+    (* value: TLA+ value to be serialized. *)
+    (* dest: Destination to serialize to such as a file or URL. *)
+    (* options: Record of serializer-specific options. *)
+    (* options must include record "ser" with a String identifying the serializer. *)
+    CHOOSE r \in [exitValue : Int, stdout : STRING, stderr : STRING] : TRUE
+
+Deserialize(src, options) ==
+    (* dest: Destination to serialize to such as a file or URL. *)
+    (* options: Record of serializer-specific options. *)
+    (* options must include record "ser" with a String identifying the serializer. *)
+    CHOOSE r \in [exitValue : Int, stdout : STRING, stderr : STRING] : TRUE
```

Comment:
This does not handle the case when deserializers deserialize to TLA+ values other than string, s.a. sets, or (generic) functions.

Reply:
Good point, updated it in 6a44bfe